app/domain/authentication/authn_k8s/validate_k8s_resource.rb

```diff
@@ -0,0 +1,64 @@
+require 'forwardable'
+require 'command_class'
+
+module Authentication
+  module AuthnK8s
+
+    ValidateK8sResource ||= CommandClass.new(
+      dependencies: {
+        k8s_resolver_class: K8sResolver,
+        k8s_object_lookup:  nil,
+        pod:                nil,
```

Comment:
what's this? if the value is nil then you can remove the dependency altogether.

Reply:
also - how come you don't need the `k8s_object_lookup` class?

Reply:
Cannot be removed.
The dependencies just don't have any default value because they need to be supplied from the outside.

Reply:
I can replace it with regular class (as I can for any other command-class).
would that make more sense?
I thought it is appropriate to use command-class here, because the flow is similar to command class.